extension/js/common/brute-force-protection.ts

```diff
@@ -0,0 +1,120 @@
+/* ©️ 2016 - present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com */
+
+import { Str } from './core/common.js';
+import { Lang } from './lang.js';
+import { Catch } from './platform/catch.js';
+import { AcctStore } from './platform/store/acct-store.js';
+
+const SUBMIT_BUTTON_SELECTOR = '.action_ok, .action_test_current_passphrase, .action_verify, .action_show_public_key';
+const PASSPHRASE_ATTEMPTS_INTRODUCE_SELECTOR = '.passphrase_attempts_introduce_label';
+const ANTI_BRUTE_FORCE_PROTECTION_ATTEMPTS_MAX_VALUE = 5;
+const BLOCKING_TIME_IN_MILI_SECONDS = 5 * 60 * 1000;
+
+export class BruteForceProtection {
+  private acctEmail!: string;
+  private failedPassPhraseAttempts!: number;
+  private lastUnsuccessfulPassphraseAttempt!: number;
+  private previousState: undefined | boolean;
+  private submitButtonText = '';
+  private readonly CHECK_BRUTE_FORCE_FREQUENCY = 1000;
+
+  public constructor(acctEmail: string) {
+    this.acctEmail = acctEmail;
+  }
+
+  public init = async () => {
+    const storage = await AcctStore.get(this.acctEmail, ['fesUrl', 'failed_passphrase_attempts', 'last_unsuccessful_passphrase_attempt']);
+    this.submitButtonText = $(SUBMIT_BUTTON_SELECTOR).text();
+    this.failedPassPhraseAttempts = storage.failed_passphrase_attempts ?? 0;
+    this.lastUnsuccessfulPassphraseAttempt = storage.last_unsuccessful_passphrase_attempt ?? 0;
+    await this.monitorBruteForceProtection();
+    Catch.setHandledInterval(() => this.monitorBruteForceProtection(), this.CHECK_BRUTE_FORCE_FREQUENCY);
+  };
+
+  public passphraseCheckFailed = async () => {
+    await AcctStore.set(this.acctEmail, {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      failed_passphrase_attempts: this.failedPassPhraseAttempts + 1,
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      last_unsuccessful_passphrase_attempt: new Date().valueOf(),
+    });
+    this.failedPassPhraseAttempts += 1;
+    this.lastUnsuccessfulPassphraseAttempt = new Date().valueOf();
```

Comment:
it can be moved to be before `await AcctStore.set` to make values the same:
```js
this.failedPassPhraseAttempts++;
this.lastUnsuccessfulPassphraseAttempt = new Date().valueOf();
await AcctStore.set(this.acctEmail, {
  // eslint-disable-next-line @typescript-eslint/naming-convention
  failed_passphrase_attempts: this.failedPassPhraseAttempts,
  // eslint-disable-next-line @typescript-eslint/naming-convention
  last_unsuccessful_passphrase_attempt: this.lastUnsuccessfulPassphraseAttempt,
});
```